### ISSUE TITLE
Text Input is not at bottom with UITabbarController

### ISSUE THREAD

**Issue by tungvoduc** [badoo/Chatto#197]
When pushing to BaseChatViewController and hide tab bar, somehow Text input view is not at bottom. See file attached.
![image](https://cloud.githubusercontent.com/assets/6437873/17539347/17d73e30-5eb6-11e6-9dc3-4650e42ef9c2.jpeg)

**Comment by diegosanchezr** [comment]
Hi @tungvoduc,

Can you try this patch?
```
diff --git a/Chatto/Source/ChatController/BaseChatViewController.swift b/Chatto/Source/ChatController/BaseChatViewController.swift
index 4bfeb2a..cdb68f6 100644
--- a/Chatto/Source/ChatController/BaseChatViewController.swift
+++ b/Chatto/Source/ChatController/BaseChatViewController.swift
@@ -146,7 +146,7 @@ public class BaseChatViewController: UIViewController, UICollectionViewDataSourc
         self.view.addConstraint(NSLayoutConstraint(item: self.inputContainer, attribute: .Top, relatedBy: .GreaterThanOrEqual, toItem: self.topLayoutGuide, attribute: .Bottom, multiplier: 1, constant: 0))
         self.view.addConstraint(NSLayoutConstraint(item: self.view, attribute: .Leading, relatedBy: .Equal, toItem: self.inputContainer, attribute: .Leading, multiplier: 1, constant: 0))
         self.view.addConstraint(NSLayoutConstraint(item: self.view, attribute: .Trailing, relatedBy: .Equal, toItem: self.inputContainer, attribute: .Trailing, multiplier: 1, constant: 0))
-        self.inputContainerBottomConstraint = NSLayoutConstraint(item: self.view, attribute: .Bottom, relatedBy: .Equal, toItem: self.inputContainer, attribute: .Bottom, multiplier: 1, constant: 0)
+        self.inputContainerBottomConstraint = NSLayoutConstraint(item: self.bottomLayoutGuide, attribute: .Top, relatedBy: .Equal, toItem: self.inputContainer, attribute: .Bottom, multiplier: 1, constant: 0)
         self.view.addConstraint(self.inputContainerBottomConstraint)
 
         let inputView = self.createChatInputView()
@@ -183,7 +183,6 @@ public class BaseChatViewController: UIViewController, UICollectionViewDataSourc
         if self.isFirstLayout {
             self.updateQueue.start()
             self.isFirstLayout = false
-            self.inputContainerBottomConstraint.constant = self.bottomLayoutGuide.length
         }
     }
 ```

If that doesn't work, can you provide an example where the bug reproduces?

**Comment by tungvoduc** [comment]
@diegosanchezr :  It is very easy to reproduce it. You just need to download sample project from dev branch, then:
- Embed in a Tab bar controller to navigation controller
- DemoChatViewController -> Hide bottom bar on push
- Run the app and choose "Chat with 10000 messages"

**Comment by diegosanchezr** [comment]
I did a quick try and run into http://stackoverflow.com/questions/26151648/bottom-layout-guide-length-issue-with-tabbar-after-pushing. Unfortunately, I don't have time to keep digging further now.

**Comment by zwang** [comment]
I got similar issue sometimes too. In my app, I have a root tab bar controller, and each tab links to a navigational controller which has own root view controller (lets name it A). I am calling A.navigationController.push(B) to go to another view controller when user click a in-app notification. It only happens when B is not yet created yet. Here B is a subclass of BaseChatViewController  Haven't got time to look into the exact cause yet. 

The work around for me is in B's `viewDidLoad` method, before calling `super.viewDidLoad()`, I added `self.tabBarController?.tabBar.hidden = true`, this seems to have fixed this issue so far. 

Hope this helps.
